### DISCUSSION TITLE
how to log python file

### DISCUSSION THREAD

**Discussion by gastonstrizzolo** [Ericsson/codechecker#3497] [Q&A] [answered]
**Describe the bug**
Hi, I have been analyzing some C files, and I have noticed how awesome your project is.
But when I try to analyze python files, I didn't find any documentation :(. I have tried this like if it were C file but obviously didn´t work:
CodeChecker log -b "python3 my_file.py" --output ./my_file.json
What I have understood about the "log" command, is that it catches compiler steps and saves it the json output. In C with gcc I can imagine how it captures compilation steps, but I was curious about how does it works with interpreted languages

**CodeChecker version**
6.17

**To Reproduce**
CodeChecker log -b "python3 my_file.py" --output ./my_file.json
But it results in an empty json

**Expected behaviour**
A complete json

**Desktop (please complete the following information)**
Ubuntu 20.04.3 LTS

**Comment by csordasmarton** [comment, marked as answer]
`CodeChecker log` can be used to catch C/C++ compiler calls (g++, gcc, clang) and it will create a compilation database. `CodeChecker analyze` will work with this compilation database and it will create analyzer result files (basically plist files) to a given directory (report directory).  These result files can be parsed on the CLI (CodeChecker parse) or can be stored to a running CodeChecker server. This approach will work only if you want to run Clang Tidy or Clang SA analyzers.

For other analyzers we created a simple tool which can be used to convert an analyzer output to CodeChecker specific output. For example if you would like to run a python specific analyzer such as pylint you can do it on the following way: https://github.com/Ericsson/codechecker/blob/master/docs/tools/report-converter.md#pylint
